enumerables.rb

```diff
@@ -0,0 +1,184 @@
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+# rubocop:disable Metrics/BlockNesting
+# rubocop:disable Metrics/ModuleLength
+module Enumerable
+  def my_each
+    if block_given?
+      i = 0
+      while i < length
+        if instance_of?(Array)
+          yield self[i]
+        else
+          yield(keys[i], values[i])
+        end
+        i += 1
+      end
+    else
+      puts 'block was not given'
+    end
+  end
```

Comment:
- [ ]  enumerables #my_each returns an Enumerator if no block is given
- [ ]  enumerables #my_each when a block is given when self is an array returns the array itself after calling the given block once for each element in self
    ` Failure/Error: expect(array.my_each(&block)).to eq(array.each(&block)) `
- [ ]  enumerables #my_each when a block is given when self is a range returns the range itself after calling the given block once for each element in self
    ``` NameError:
       undefined local variable or method `length' for 5..50:Range
   ```
- [ ] enumerables #my_each when a block is given when self is a range behaves like #each calls the given block once for each element in self
    ```  NameError:  undefined local variable or method `length' for 5..50:Range ```
- [ ]  enumerables #my_each when a block is given when self is a hash behaves like #each calls the given block once for each element in self
  ```  expected: "[:a, 1][:b, 2][:c, 3][:d, 4][:e, 5]" but  got: "abcde" ```